src/components/ReportUpload/CounterUpload/CounterUpload.test.js

```diff
@@ -172,7 +184,8 @@ describe('CounterUpload', () => {
     {
       name: 'error 404 without response body',
       mockFile: file,
-      expectedError: expect.stringContaining('An unexpected error has occurred'),
+      expectedError: 'An error has occurred.',
+      expectedDetails: 'invalid json response body at',
```

Comment:
Use a regular expression for partial matching and keep the exact matching in the test function. This preserves exact matches in other test cases.

```suggestion
      expectedDetails: /invalid json response body at/,
```